### PR TITLE
Refactor description to not rely on Arc instance.

### PR DESCRIPTION
**Pull request by mmandlis**
next step in https://github.com/PolymerLabs/arcs/issues/2402

### CODE REVIEW

File: src/runtime/description-formatter.ts [hunk 1]
```diff
@@ -11,42 +11,35 @@
...
-  
-  constructor(arc: Arc, relevance?: Relevance) {
-    assert(arc, `Arc is mandatory`);
-    this.arc = arc;
-    this.relevance = relevance;
-  }
 
-  async getDescription(recipe: {patterns: string[], particles: Particle[]}) {
-    await this._updateDescriptionHandles();
+  constructor(private readonly particleDescriptions = <ParticleDescription[]>[],
```
Comment (lindner) [inline src/runtime/description-formatter.ts]: Shane had some opinions on whether we should instantiate properties in the constructor like this.

I'm fine with it fwiw, as it's a more compact representation...

**Comment by lindner** [comment]
Also, let's submit this before my typescriptification of the mock composer classes.  My rebase will be easier.

File: src/runtime/description-formatter.ts (hunk 1 above)
Comment (mmandlis) [reply src/runtime/description-formatter.ts]: I relied on some existing examples of ctors, 
but I can update it in a follow up PR, if it is an undesirable way.

**Comment by mmandlis** [comment]
> Also, let's submit this before my typescriptification of the mock composer classes. My rebase will be easier.

sg, submitting now.

**Pull request merged**